dataproc/snippets/create_cluster_test.py

```diff
@@ -53,8 +63,21 @@ def test_cluster_create(capsys):
     # Wrapper function for client library function
     try:
         create_cluster.create_cluster(PROJECT_ID, REGION, CLUSTER_NAME)
+        out, _ = capsys.readouterr()
+        assert CLUSTER_NAME in out
+    except AlreadyExists:
+        request = dataproc.GetClusterRequest(project_id=PROJECT_ID, region=REGION, cluster_name=CLUSTER_NAME)
+        response = cluster_client.get_cluster(request=request)
+        assert response.status.state == dataproc.ClusterStatus.State(2)  # verify the cluster is in the RUNNING state
+
+        status_start = response.status.state_start_time  # when the cluster started being in the RUNNING state
+        now = datetime.datetime.now(datetime.timezone.utc)
+        diff = now-status_start
+        # check that it's been running for less than 20 min
+        # this means we probably did a backoff during the creation for a different reason
+        # and that this cluster was created as part of this test, so we can continue
+        assert diff.seconds/60 < 20
```

Comment:
Does the cluster name include a unique identifier? If so, wouldn't that be enough to make sure the cluster is from this test process?

Reply:
The chances of them colliding would be very low, it's a UUID4 so it should not be an issue. But I also know we're truncating them. Anyways, I don't have a strong opinion. Removing it might be cleaner and simpler, but having the extra check doesn't hurt. I guess the only thing is if it's worth maintaining the extra complexity.

Reply:
Wdyt about removing the time check but keeping the check to see if it's in the running state? That might be the happy medium of doing some due diligence